lib/src/main/java/org/team1540/rooster/drive/pipeline/ProfileInput.java

```diff
@@ -0,0 +1,85 @@
+package org.team1540.rooster.drive.pipeline;
+
+import edu.wpi.first.wpilibj.Timer;
+import java.util.OptionalDouble;
+import org.jetbrains.annotations.NotNull;
+import org.team1540.rooster.functional.Input;
+import org.team1540.rooster.motionprofiling.MotionProfile;
+import org.team1540.rooster.motionprofiling.MotionProfile.Point;
+
+/**
+ * Class to get commanded drive values from a set of motion profiles.
+ *
+ * This class is non-reusable; the first call to {@link #get()} begins a timer to determine where
+ * the input is in the profile. To execute multiple profiles, create multiple {@code
+ * ProfileInputs}.
+ */
+public class ProfileInput implements Input<TankDriveData> {
+
+  private MotionProfile left;
+  private MotionProfile right;
+
+  private Timer timer = new Timer();
+
+  private boolean finished;
+
+  /**
+   * Create a new {@code ProfileInput}.
+   *
+   * @param left The left-side profile to execute.
+   * @param right The right-side profile to execute.
+   */
+  public ProfileInput(@NotNull MotionProfile left, @NotNull MotionProfile right) {
+    this.left = left;
+    this.right = right;
+  }
+
+  @NotNull
+  private Point getCurrentSegment(@NotNull MotionProfile trajectory, double currentTime) {
+    // Start from the current time and find the closest point.
+    int startIndex = Math.toIntExact(Math.round(currentTime / trajectory.get(0).dt));
```

Comment:
ah but what if the dt isn't the same across? Technically the motion profiles don't prescribe that's a requirement.

Also do we want to be matching based on time or based on position?

Reply:
* We're assuming the DT is the same across because, well, we are.

* Based on time IMO—based on position could lead to all sorts of weird things